libpod/container.go

```diff
@@ -1590,3 +1590,113 @@ func (c *Container) Sync() error {
 
 	return nil
 }
+
+type resolv struct {
+	nameServers   []string
+	searchDomains []string
+	options       []string
+}
+
+// generateResolvConf generates a containers resolv.conf
+func (c *Container) generateResolvConf() (string, error) {
+	// Copy /etc/resolv.conf to the container's rundir
+	resolvPath := "/etc/resolv.conf"
+
+	// Check if the host system is using system resolve and if so
+	// copy its resolv.conf
+	_, err := os.Stat("/run/systemd/resolve/resolv.conf")
+	if err == nil {
+		resolvPath = "/run/systemd/resolve/resolv.conf"
+	}
+	orig, err := ioutil.ReadFile(resolvPath)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to read %s", resolvPath)
+	}
+	if len(c.config.DNSServer) == 0 && len(c.config.DNSSearch) == 0 && len(c.config.DNSOption) == 0 {
+		return c.WriteStringToRundir("resolv.conf", fmt.Sprintf("%s", orig))
+	}
+
+	// Read and organize the hosts /etc/resolv.conf
+	resolv := createResolv(string(orig[:]))
+
+	// Populate the resolv struct with user's dns search domains
+	if len(c.config.DNSSearch) > 0 {
+		resolv.searchDomains = nil
+		// The . character means the user doesnt want any search domains in the container
+		if !StringInSlice(".", c.config.DNSSearch) {
+			resolv.searchDomains = append(resolv.searchDomains, c.Config().DNSSearch...)
+		}
+	}
+
+	// Populate the resolv struct with user's dns servers
+	if len(c.config.DNSServer) > 0 {
+		resolv.nameServers = nil
+		for _, i := range c.config.DNSServer {
+			resolv.nameServers = append(resolv.nameServers, i.String())
+		}
+	}
+
+	// Populate the resolve struct with the users dns options
+	if len(c.config.DNSOption) > 0 {
+		resolv.options = nil
+		resolv.options = append(resolv.options, c.Config().DNSOption...)
+	}
+	return c.WriteStringToRundir("resolv.conf", resolv.ToString())
+}
+
+// createResolv creates a resolv struct from an input string
+func createResolv(input string) resolv {
+	var resolv resolv
+	for _, line := range strings.Split(input, "\n") {
+		if strings.HasPrefix(line, "search") {
+			fields := strings.Fields(line)
```

Comment:
Might be worth checking if fields is at least 2 elements long so we don't segfault on the next line if we parse a badly formed resolv.conf? Can just ignore if the line is bad

Reply:
fixed